### ISSUE TITLE
CKAN crashes during "Refresh" if packages have unmet deps

### ISSUE THREAD

**Issue by XANi** [KSP-CKAN/CKAN#1065]
Error message: https://gist.github.com/XANi/f3b680f9fc94a288e61a

version `v1.6.20-0-g3c4f781 (beta)` (on linux)

I had some problems with modules (near future solar power not interacting with "normal" power) so I decided to save my plugin list into ckan file, clean plugins folder, validate files in steam and tried to load plugins ( https://gist.github.com/XANi/735b1c1467aee586d5e4) from plugin list.

It installed a few and then complained about not being able to find ModuleManager in repo so I though maybe I have old repo version, clicked "Refresh" and it crashed. Now everytime I click refresh it crashes

**Comment by Dazpoet** [comment]
Sorry for being slow in responding to this and I assume it might have been resolved in some other way by now but... did you uninstall the files through CKAN or did you just flat out clean out the GameData folder? Doing the latter will have CKANs registry still belive that you have mods installed and that would lead to the issue described above.

**Comment by XANi** [comment]
I had to clean it because CKAN didnt let me uninstall some mods,

What I basically did was exporting every installed mod into .ckan, then cleaned up both ckan and KSP dirs and tried to install from that.

But for some reason the mod list was incomplete the first time it downloaded and I had mods requiring Module Manager but no Module Manager on list and every time I clicked refresh it crashed  with `CommunityTechTree requires ModuleManager but nothing provides it`, so it never could get a "right" repo metadata

After clearing up again and downloading full repository metadata same steps worked correctly.

It kinda looked like "Refresh" checked deps before refreshing metadata from repo, while it probably should download latest version of database first.
